Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -256,6 +263,40 @@ void AppendPipeline::dispatchDecryptionKey(GstBuffer* buffer)
 }
 #endif
 
+void AppendPipeline::setStartupBufferingComplete(bool complete)
+{
+    LockHolder locker(m_newSampleLock);
+    m_startupBufferingComplete = complete;
+}
+
+void AppendPipeline::flushStartupSamples()
+{
+    if (!enableStartupBuffering() || !m_playerPrivate)
+        return;
+
+    Vector< GRefPtr<GstSample> > samples;
+    {
+        LockHolder locker(m_newSampleLock);
```

Comment:
This can be done with locker.unlockEarly() instead of using a code block.

Reply:
done